src/main/scala/kamon/newrelic/metrics/NewRelicDistributionMetrics.scala

```diff
@@ -46,8 +47,8 @@ object NewRelicDistributionMetrics {
   }
 
   private def makePercentiles(name: String, end: Long, distValue: Distribution, instrumentBaseAttributes: Attributes): Seq[Metric] = {
-    val kamonPercentiles: Seq[Distribution.Percentile] = distValue.percentiles
-    kamonPercentiles.map { percentile =>
+    val kamonPercentiles: Seq[Distribution.Percentile] = percentilesToReport.map(rank => distValue.percentile(rank))
+    kamonPercentiles.filter(percentileValue => percentileValue != null).map { percentile =>
```

Comment:
I don't think the local `val` is helping us here anymore.  Let's ditch it and do something like
```
percentilesToReport.map( ...)
  .filter(not null)
  .map(to Metric)
```

Reply:
that's fair. will change.